core/functions/root_object.go

```diff
@@ -22,8 +22,39 @@ import (
 
 	"github.com/dolthub/doltgresql/core/id"
 	"github.com/dolthub/doltgresql/core/rootobject/objinterface"
+	pgtypes "github.com/dolthub/doltgresql/server/types"
 )
 
+// DeserializeRootObject implements the interface objinterface.Collection.
+func (pgf *Collection) DeserializeRootObject(ctx context.Context, data []byte) (objinterface.RootObject, error) {
+	return DeserializeFunction(ctx, data)
+}
+
+// DiffRootObjects implements the interface objinterface.Collection.
+func (pgf *Collection) DiffRootObjects(ctx context.Context, ours, theirs, ancestor objinterface.RootObject) ([]objinterface.RootObjectDiff, error) {
+	// TODO: these are example diffs for the sake of output comparison
+	return []objinterface.RootObjectDiff{
+		{
+			Type:          pgtypes.Text,
+			FieldName:     "return_type",
```

Comment:
Interesting to see the diff be so granular. I was having a hard time thinking of when you'd want to merge just the return type or language, but not the body of the function. Do you see concrete cases where people would merge only part of a function like that?